src/BTCPayServer.Lightning.Common/PayResponse.cs

```diff
@@ -9,6 +11,7 @@ public enum PayResult
 
     public class PayResponse
     {
+        [JsonConstructor]
```

Comment:
I always thought json ctors needed to be parameterless

Reply:
I'm not sure about this and what behavior it changes

Reply:
It has to be either an empty standard constructor or marked with this attribute. I chose to use the attribute as the result needs to be available in all cases.

I think we didn't run across the serialization of this class before, because I wasn't handling the responses correctly in LNbank. That's the second part I fixed in this PR. I came across it, because I saw that dennisreimann/btcpayserver#17 was prevented by an exception being thrown when serializing the `PayResponse`.

Reply:
Based on https://stackoverflow.com/questions/23017716/json-net-how-to-deserialize-without-using-the-default-constructor, I think it should work but honestly would prefer a parameterless ctor to not have to worry about edge-cases.

Reply:
Updated in [074d6ae](https://github.com/btcpayserver/BTCPayServer.Lightning/pull/78/commits/074d6ae9224e5950a5da8458583b417317eb88c8).